lib/resources/http.rb

```diff
@@ -22,23 +22,28 @@ class Http < Inspec.resource(1)
         its('Content-Length') { should cmp 258 }
         its('Content-Type') { should cmp 'text/html; charset=UTF-8' }
       end
-
-      # properly execute the HTTP call on the scanned machine instead of the
-      # machine executing InSpec. This will be the default behavior in InSpec 2.0.
-      describe http('http://localhost:8080', enable_remote_worker: true) do
-        its('body') { should cmp 'local web server on target machine' }
-      end
     "
 
     def initialize(url, opts = {})
       @url = url
       @opts = opts
 
-      if use_remote_worker?
-        return skip_resource 'curl is not available on the target machine' unless inspec.command('curl').exist?
-        @worker = Worker::Remote.new(inspec, http_method, url, opts)
-      else
+      # Prior to InSpec 2.0 the HTTP test had to be instructed to run on the
+      # remote target machine. This warning will be removed.
+      if opts.key?(:enable_remote_worker) && !inspec.local_transport?
+        warn 'Ignoring `enable_remote_worker` option, the `http` resource ',
+             'remote worker is enabled by default for remote targets and ',
+             'cannot be disabled'
+      end
+
+      # Run locally if InSpec is ran locally and remotely if ran remotely
+      if inspec.local_transport?
         @worker = Worker::Local.new(http_method, url, opts)
+      elsif !inspec.command('curl').exist?
```

Comment:
Code-flow-wise, it would be way cleaner to couple this check with the remote worker itself. I initially did not do that because we didn't have the ability to raise exceptions to skip resources and you could only really call `skip_resource` during initialize. But now, since we're here, we can clean this up and this `if` statement will read much cleaner to the new user.

I would change this whole conditional to look like this:

```ruby
if inspec.local_transport?
  @worker = Worker::Local.new(http_method, url, opts)
else
  @worker = Worker::Remote.new(inspec, http_method, url, opts)
end
```

And then move the `curl` check to the initialize of the `Worker::Remote` class. Make sense?

Reply:
Yup, that is much cleaner! Thanks for the historical context. I will make it happen.